src/osd/OSD.h

```diff
@@ -1453,7 +1453,7 @@ class OSD : public Dispatcher,
 
   void clear_waiting_sessions() {
     Mutex::Locker l(session_waiting_lock);
-    for (map<spg_t, set<Session*> >::iterator i =
+    for (map<spg_t, set<Session*> >::const_iterator i =
```

Comment:
this *creates* a `const_iterator` from an `iterator`. which incurs the cost of calling a copy ctor.  if you want to make sure that we are using the `const_iterator` for reading containers. maybe you could use `auto`, `session_waiting_for_pg.cbegin()`, `session_waiting_for_pg.cend()` instead?